src/reader/ft_split_smart.c

```diff
@@ -0,0 +1,120 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split_smart.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: al7aro <marvin@42.fr>                      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/09/14 11:32:13 by al7aro            #+#    #+#             */
+/*   Updated: 2022/09/19 12:08:09 by al7aro-g         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#include "libft.h"
+#include "tab.h"
+
+int	is_special(char *str)
+{
+	int	i;
+
+	i = 0;
+	if (*str == '<')
```

Comment:
also always put the constant on the left
if (*str == '<') -> if ('<' == *str)
it helps to avoid silly mistakes like:
if (*str **=** '<')